apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/add-partner-form.client.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+import { useAddPartner } from "../../hooks/use-add-partner";
+
+const formSchema = z.object({
+  name: z.string().min(1, {
+    message: "Partner name is required",
+  }),
+  domains: z.string().optional(),
+  bundleIds: z.string().optional(),
+});
+
+export function AddPartnerForm({ ecosystemId }: { ecosystemId: string }) {
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema.partial()),
+  });
+
+  const { addPartner, isLoading } = useAddPartner({
+    onSuccess: () => {
+      form.reset();
+    },
+    onError: (error) => {
+      const message =
+        error instanceof Error
+          ? error.message
+          : "Failed to add ecosystem partner";
+      toast.error(message);
+    },
+    mutationKey: ["addPartner", ecosystemId],
+  });
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit((values) =>
+          addPartner({
+            ecosystemId: ecosystemId,
+            name: values.name,
+            allowlistedDomains: values.domains?.split(/,| /) ?? [],
+            allowlistedBundleIds: values.bundleIds?.split(/,| /) ?? [],
+            permissions: "PROMPT_USER_V1",
+          }),
+        )}
+        className="flex space-x-2"
```

Comment:
prefer "gap" over "space"

https://twitter.com/adamwathan/status/1663527846319407105

Reply:
(obviously applies everywhere you are currently using "space")